src/Shipment.php

```diff
@@ -0,0 +1,352 @@
+    {
+        // TODO convert to shipment API
+        if (empty($to)) {
+            $to = date('YmdHs');
+        }
+        $query_params = "from={$from}&to{$to}";
+
+        if (!empty($tracking_pin)) {
+            $query_params = "trackingPIN={$tracking_pin}";
+        }
+
+        $response = $this->get(
+            "rs/{$this->config['customer_number']}/{$this->config['customer_number']}/shipment?" . $query_params,
+            ['Accept' => 'application/vnd.cpc.shipment-v8+xml'],
+            $options
+        );
+        return $response;
+
+    }
+
+    public function requestShipmentRefund($shipment_id, $email, $options) {
+        $content = [
+            'email' => $email
+        ];
+
+        $xml = Array2XML::createXML('non-contract-shipment-refund-request', $content);
+        $envelope = $xml->documentElement;
+        $envelope->setAttribute('xmlns',
+            'http://www.canadapost.ca/ws/shipment-v8');
+        $payload = $xml->saveXML();
+        $response = $this->post(
+            "rs/{$this->config['customer_number']}/{$this->config['customer_number']}/shipment/{$shipment_id}/refund",
+            [
+                'Content-Type' => 'application/vnd.cpc.shipment-v8+xml',
+                'Accept' => 'application/vnd.cpc.shipment-v8+xml',
+            ],
+            $payload,
+            $options
+        );
+        return $response;
+    }
+
+    /**
+     * Transmit shipments for pickup by Canada Post.
+     *
+     * @param array $manifest_address
+     *   The destination info.
+     *   <code>
+     *     $manifest_address = [
+     *       'manifest-company' => 'ACME Inc.',
+     *       'phone-number'     => '778 867 5309',
+     *       'address-details'  => [
+     *         'address-line-1'  => '123 Main St',
+     *         'city'            => 'Kelowna',
+     *         'prov-state'      => 'BC',
+     *         'country-code'    => 'CA',
+     *         'postal-zip-code' => 'V1X 1M2',
+     *       ],
+     *     ]
+     *   </code>
+     * @param array $group_ids
+     *   The group IDs. The Transmit Shipments service will create a manifest
+     * for each group. The manifest will list the shipments included in the
+     * group.
+     * @param array $options
+     *   The options to pass along to the Guzzle Client.
+     *
+     * @see https://www.canadapost.ca/cpo/mc/business/productsservices/developers/services/shippingmanifest/transmitshipments.jsf
+     * for all available options for the sender,destination and parcel params.
+     * @return \DOMDocument
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function transmitShipments(
+        array $manifest_address,
+        array $group_ids,
+        array $options = []
+    ) {
+        $this->verifyPostalCode($manifest_address);
+        $content = [
+            'group-ids' => [
+                'group-id' => $group_ids
+            ],
```

Comment:
Same optional syntax correction, either do `'group-ids' => ['group-id' => $group_ids]` or add a comma at the end of the element.